### PR TITLE
adjust locationlog to have its own voip column

### LINKED ISSUE #3677: Large join for locationlog views just for VoIP flag
The locationlog table has to be joined to the node table in order to know the VoIP status of a device and filter it when viewing locationlog entries.

So when the node table contains multiple hundred thousands entries, then this creates a massive join.

Also, if you change the status of a node to make it voip, then the locationlog entry will say that its voip while the locationlog entry actually represents a non-voip reply.

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Prevent joining the locationlog to the node table to obtain the voip info

# Impacts
locationlog view/create

# Issue
fixes #3677 

# Delete branch after merge
YES

# NEWS file entries
## Bug Fixes
* Prevent costly join with node table when searching latest locationlog entries for a node (#3677)

### CODE REVIEW

**Comment by julsemaan** [comment]
@satkunas, @jrouzierinverse, @fdurand 

Could you guys take a look at this solution and let me know if there will be objection to the integration of this fix before I have it applied at my customer deployment.

**Comment by jrouzierinverse** [comment]
Calls to locationlog_insert_start do not include voip.

**Comment by julsemaan** [comment]
I'm not actually fully done, haven't even tested if it compiles yet :P

Just wanted to know if anybody objects to the general idea before putting more energy into it

**Comment by jrouzierinverse** [comment]
Why not just add the tenant_id in the join criteria for node and location in maintenance since that would fix the issue with out any database changes.
Now that fix can be a +1.

**Comment by julsemaan** [comment]
that would be perfect for fixing the performance issue in an immediate manner and we'll backlog the other improvement (having the latest voip state) for 8.3 and not rush it

**Comment by extrafu** [comment]
Please rebase and let's get this in v9 as it's easier to get schema changes in big fat releases.

**Comment by julsemaan** [comment]
PF v9 codename: My big fat greek release

**Comment by extrafu** [comment]
Rebase the gyros @jrouzierinverse

**Comment by julsemaan** [comment]
Just rebased it and added some of them garlic potatos with that

**Comment by extrafu** [comment]
@jrouzierinverse Please review and merge ASAP.

File: db/upgrade-X.X.X-X.Y.Z.sql [hunk 1]
```diff
@@ -100,4 +100,21 @@ DROP TABLE violation;
...
+-- Update the locationlog voip information from the node table information
+--
+update locationlog join node on locationlog.mac = node.mac set locationlog.voip = node.voip;
+
+--
+-- Add potd column in person table
+--
+
+ALTER TABLE person
+    ADD `potd` enum('no','yes') NOT NULL DEFAULT 'no';
```
Comment (jrouzierinverse) [inline db/upgrade-X.X.X-X.Y.Z.sql]: I am not sure since belongs here.

No change to the pf-schema-X.Y.Z.sql

**Comment by jrouzierinverse** [comment]
t/serialized_unittests/db/apply_upgrade.t failed.

**Comment by julsemaan** [comment]
apply upgrade now works fine

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]
VOIP status is never passed into the locationlog_insert_start function.

https://github.com/inverse-inc/packetfence/blob/2536d0df5057fd4b4fe061ccd77a761d217440c4/lib/pf/locationlog.pm#L387
https://github.com/inverse-inc/packetfence/blob/2536d0df5057fd4b4fe061ccd77a761d217440c4/lib/pf/locationlog.pm#L433

**Comment by julsemaan** [comment]
> VOIP status is never passed into the locationlog_insert_start function.
> 
> [packetfence/lib/pf/locationlog.pm](https://github.com/inverse-inc/packetfence/blob/2536d0df5057fd4b4fe061ccd77a761d217440c4/lib/pf/locationlog.pm#L387)
> 
> Line 387 in [2536d0d](/inverse-inc/packetfence/commit/2536d0df5057fd4b4fe061ccd77a761d217440c4)
> 
>  locationlog_insert_start($switch, $switch_ip, $switch_mac, $ifIndex, $vlan, $mac, $connection_type, $connection_sub_type, $user_name, $ssid, $stripped_user_name, $realm, $role, $locationlog_mac, $ifDesc); 
> 
> [packetfence/lib/pf/locationlog.pm](https://github.com/inverse-inc/packetfence/blob/2536d0df5057fd4b4fe061ccd77a761d217440c4/lib/pf/locationlog.pm#L433)
> 
> Line 433 in [2536d0d](/inverse-inc/packetfence/commit/2536d0df5057fd4b4fe061ccd77a761d217440c4)
> 
>  locationlog_insert_start($switch, $switch_ip, $switch_mac, $ifIndex, $vlan, $mac, $connection_type, $connection_sub_type, $user_name, $ssid, $stripped_user_name, $realm, $role, undef, $ifDesc)

Good catch, just fixed it

**Review by jrouzierinverse — APPROVED** [review]

**Pull request merged**